### PR TITLE
add .zip artifacts for macOS

### LINKED ISSUE #221: Use .zip and .dmg for packaging on macOS
### Goal

Use .zip and .dmg (?) for packaging on macOS while keeping .tar.gz's for backwards compatibility

### Motivation

More commonly used formats for macOS app distribution.

### Specification

* Change the CI builds to also publish .zip's and .dmg's, which should be easy on the macOS runner
* Add multiple format downloads on the website
* Change updater to use other formats?

### Drawbacks

Not really.

### Unresolved Questions

No

### Alternatives Considered

No

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by ryanccn**
because yes

closes #221

### CODE REVIEW

**Review by DioEgizio — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -355,6 +356,13 @@ jobs:
         with:
           name: PrismLauncher-${{ matrix.name }}-${{ env.VERSION }}-${{ inputs.build_type }}
           path: PrismLauncher.tar.gz
+      
+      - name: Upload binary zip (macOS)
+        if: runner.os == 'macOS'
+        uses: actions/upload-artifact@v3
+        with:
+          name: PrismLauncher-${{ matrix.name }}-${{ env.VERSION }}-${{ inputs.build_type }}-zip
+          path: PrismLauncher.zip
```
Comment (DioEgizio) [inline .github/workflows/build.yml:365]: Wait... wouldn't this make a zip of a zip? Also you should update the release workflow

**Review by flowln — APPROVED** [review]

**Review by Scrumplex — CHANGES_REQUESTED** [review]
As I already wrote in the other thread:
.zip and .tar.gz work identically on macOS. I don't see a reason we would need both

**Comment by ryanccn** [comment]
.zip is generally more commonly used for distribution of macOS apps, so it would be more familiar to users I guess

**Comment by TheKodeToad** [comment]
> .zip is generally more commonly used for distribution of macOS apps, so it would be more familiar to users I guess

Afaik most mac users know what .tar.gz is, and double clicking on it (what most users would start with) extracts it anyway.

**Comment by ryanccn** [comment]
Also, most software are distributed in multiple formats (at least for macOS apps this is very often the case) in case one format is unavailable or something. Generally nicer to have more artifacts/formats available IMO

**Comment by Scrumplex** [comment]
So what's the game plan? I am up to distributing dmgs, but zips seem redundant to me if we already have tars.

**Comment by ryanccn** [comment]
dmgs are quite slow and they're not incredibly useful

**Pull request closed without merging**